webapp/src/DOMJudgeBundle/Service/ScoreboardService.php

```diff
@@ -347,10 +351,8 @@ public function calculateScoreRow(Contest $contest, Team $team, Problem $problem
             // if correct, don't look at any more submissions after this one
             if ($judging->getResult() == Judging::RESULT_CORRECT) {
                 $correctJury = true;
-                $timeJury    = $submitTime;
                 if (!$submission->isAfterFreeze()) {
                     $correctPubl = true;
-                    $timePubl    = $submitTime;
                 }
                 // stop counting after a first correct submission
```

Comment:
It took me a while to find this comment/break to understand why it's correct (which is obvious in hindsight). Perhaps it's easier to grok if that is announced at the loop header that we only count until the first correct submission.

Reply:
Moving the code that sets `$time{Jury,Publ}` outside of the `if ($judging->getResult() == Judging::RESULT_CORRECT)` condition means that it is triggered by *all* submissions. Unless I misunderstand something, we want this to be set only for potentially correct submissions, so those that are either correct or still pending.

Reply:
It does not really matter. This function is just about filling the cache. This column only has relevance when there is a pending or correct submission.

Before this change, it would be set to int `0` when a problem was not yet solved.

Reply:
I still don't quite agree. What if there are 2 submissions, say the first pending and the second correct (to take an extreme case). In that case the time recorded using `$timeJury` should be that of the first, pending submission if it is to be interpreted as the earliest potential solve time. But in the current loop (assuming it's ordered by submission id/time) it would be that of the second submission.
How is `$timeJury` that is stored in `scorecache.solvetime_restricted` used later?

Reply:
This case indeed futher complicates the issue.

Would it also mean that if there's indeed a submission judged correct received at 20 but a pending one received at 18, we should also not show the score value of "20" on the scoreboard at all?